### DISCUSSION TITLE
Tool Only Runs on Windows (not Linux or macOS)

### DISCUSSION THREAD

**Discussion by wherron01** [fayaz12g/totk-aar#40] [Ideas]
As this is all Python, could it be made cross-platform and a Linux build shipped?  I'd really like to use this mod tool, but I don't have a Windows install.

**Comment by morceaudebois** [comment]
Same for Mac.

**Comment by notoriousjere** [comment]
given you have python 3.11 and all of the packages used in the project, you could just run `python3 GUI.py` on either of those platforms and it should work

**Comment by wherron01** [comment]
This fails with the following error

```
Traceback (most recent call last):
  File "/home/user/totk-aar/script.py", line 21, in <module>
    file_names = os.listdir(blyt_folder)
                 ^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/totk-aar/AAR MOD/temp/Common.Product.110.Nin_NX_NVN/blyt'
Exception in thread Thread-4 (create_full):
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/home/willow/totk-aar/GUI.py", line 168, in create_full
    os.remove(blarc_file_path)
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/totk-aar/AAR MOD/temp/Common.Product.110.Nin_NX_NVN.blarc'
```

**Comment by wherron01** [comment]
I traced the issue back to the zstandard python package.  The one packaged with the mod tool does not work, and I already had the zstandard package on my install, but it defaulted to the one packaged.  Renaming it to zstandard-backup let the tool use my system package, which worked.

**Comment by notoriousjere** [comment]
There should really be a version of this project that uses poetry to manage a virtual environment that grabs the necessary dependencies instead of including them in the repo

**Comment by fayaz12g** [comment]
Currently when I use it on mac i use parallels, I know this is not preferred than running natively, especially as it is a UNIX based OS with Python already installed. I will likely release an appimage eventually that does the same thing, but works on macOS/steamOS/Linux. Unsure why the zstandard dependencies packed in are only working on windows.

**Comment by wherron01** [comment]
Just for the record, I later discovered that the generated mod wasn’t working, despite the tool not throwing any errors, and that this was because the same thing was happening with the keystone package as was with the zstandard package.

**Comment by fayaz12g** [comment]
I appreciate you updating me with that. It seems this tool does not like to work with any operating system other than windows at the moment. I'll work to fix it as a stretch goal

**Comment by dcbishop** [comment]
It works using Wine

**Comment by Samueru-sama** [comment]
> It works using Wine

How do you use this with wine? The program opens and I can input the parameters, but when I hit generate nothing happens.

**Comment by fayaz12g** [comment]
@Samueru-sama Did you specify the output folder to where you want it to go?

**Comment by fayaz12g** [comment]
I have converted this from an issue to an idea/feature request, as creating a version of this program to work on macOS and Linux is a whole separate task rather than an issue with the executable. It seems it works on wine, but even then, the default yuzu and ryujinx folders will be different and a lot of the ways this program works will have to change.

**Comment by SaltSouls** [comment]
So I have tried to use this program recently, even through wine, and so far nothing seems to work. Even when specifying a output folder, I get an error on trying to generate.

I should note it does create the folders, but fails at what I assume in the actual mod generation.

**Reply by SaltSouls** [reply]
The error in question is:
```
Error: [WinError 87] Invalid parameter: '(originated from NtQuerySystemInformation)'
```

**Comment by fayaz12g** [comment]
@SaltSouls are you trying macos, steamos or another form of linux? My guess is some of the commands it uses to get the current user and download temp files before shifting them is causing issues. I'm sorry I never got to making it work for Linux, I was going to but it was a bit of a big undertaking for me

**Reply by SaltSouls** [reply]
I'm running Arch Linux.
